code/modules/antagonists/morph/morph_antag.dm

```diff
@@ -1,6 +1,58 @@
 /datum/antagonist/morph
 	name = "Morph"
 	show_name_in_check_antagonists = TRUE
-	show_in_antagpanel = FALSE
+	show_to_ghosts = TRUE
+	job_rank = ROLE_MORPH
+	antagpanel_category = "Morph"
+	show_name_in_check_antagonists = TRUE
+
+//It does nothing! (Besides tracking)//Scratch that, it does something now at least
+
+/datum/antagonist/morph/on_gain()
+	forge_objectives()
+	return ..()
+
+/datum/antagonist/morph/greet()
+	owner.announce_objectives()
+
+/datum/antagonist/morph/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	//Give traitor appearence on hud (If they are not an antag already)
+	var/datum/atom_hud/antag/traitorhud = GLOB.huds[ANTAG_HUD_MORPH]
+	traitorhud.join_hud(owner.current)
+	if(!owner.antag_hud_icon_state)
+		set_antag_hud(owner.current, "morph")
+
+/datum/antagonist/morph/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	//Clear the hud if they haven't become something else and had the hud overwritten
+	var/datum/atom_hud/antag/traitorhud = GLOB.huds[ANTAG_HUD_MORPH]
+	traitorhud.leave_hud(owner.current)
+	if(owner.antag_hud_icon_state == "morph")
+		set_antag_hud(owner.current, null)
+
+/datum/antagonist/morph/proc/forge_objectives()
+	var/datum/objective/eat_everything/consume = new
+	consume.owner = owner
+	objectives += consume
+
+/datum/objective/eat_everything
+	explanation_text = "Eat everything and anything to sate your never-ending hunger."
+	completed = TRUE
 
-//It does nothing! (Besides tracking)
+/datum/antagonist/morph/admin_add(datum/mind/new_owner,mob/admin)
+	var/mob/living/M = new_owner.current
+	if(alert(admin,"Transform the player into a morph?","Species Change","Yes","No") == "Yes")
```

Comment:
`if(alert() != "Yes") return`

Reply:
No no no I can't do an early return here, I need the code below it to still work.

Reply:
Ah wait nvm that was an issue for a different proc.